bitpacker/src/bitpacker.rs

```diff
@@ -161,23 +162,23 @@ impl BitUnpacker {
         // so highway start is the closest multiple of 8 that is >= start_idx.
         let entrance_ramp_len = 8 - (start_idx % 8) % 8;
 
-        let highway_start: u32 = start_idx + entrance_ramp_len;
+        let highway_start: usize = start_idx + entrance_ramp_len;
 
-        if highway_start + BitPacker1x::BLOCK_LEN as u32 > end_idx {
+        if highway_start + BitPacker1x::BLOCK_LEN > end_idx {
             // We don't have enough values to have even a single block of highway.
             // Let's just supply the values the simple way.
             get_batch_ramp(start_idx, output);
             return;
         }
 
-        let num_blocks: u32 = (end_idx - highway_start) / BitPacker1x::BLOCK_LEN as u32;
+        let num_blocks: usize = (end_idx - highway_start) / BitPacker1x::BLOCK_LEN;
 
         // Entrance ramp
-        get_batch_ramp(start_idx, &mut output[..entrance_ramp_len as usize]);
+        get_batch_ramp(start_idx, &mut output[..entrance_ramp_len]);
 
         // Highway
-        let mut offset = (highway_start * self.num_bits) as usize / 8;
-        let mut output_cursor = (highway_start - start_idx) as usize;
+        let mut offset = (highway_start * self.num_bits) / 8;
```

Comment:
nit: this function might be a bit easier to read if the widening from u32 to usize was performed only when converting to bits:
```
// keep indexes as u32 and do
let mut offset = (highway_start as usize * self.num_bits) / 8;
```